documentation/content/developmentworkflow.rst

```diff
@@ -0,0 +1,115 @@
+========================
+  Development workflow
+========================
+
+.. contents::
+   :depth: 1
+   :local:
+   :backlinks: none
+
+.. highlight:: console
+
+kw is developed using Git and is hosted at Github. If you want to understand
+our development workflow better, clone the repository to have a local copy::
+
+  git clone https://github.com/kworkflow/kworkflow.git
+
+Install
+-------
+Installing kw is one of the first steps to understanding its development
+workflow. After cloning the repository, read the page related to
+:ref:`installing and removing<install-and-remove-kw>`.
+
+Tests
+-----
+To make sure our code is working as it is supposed to, we add
+:ref:`tests<tests>` to check that.
+
+All our tests can be found in the `tests` folder. Notice that every test file
+name ends with test before the .sh extension (i.e. `mk_test.sh` and
+`statistics_test.sh`). This is done so that the `run_tests.sh` file can detect
+those tests and run or list them.
+
+Unit tests
+==========
+Unit testing is an essential part of our development workflow and we use the
+`shUnit2 <https://github.com/kward/shunit2/>`_ framework to write our tests.
+Ideally, every new function should have a test related to it. So, if you're
+working on a new feature, added a new function, or changed the behavior of an
+existing one, it's fundamental to make the necessary ajustments related to
+testing.
+
+To see how to run individual test files, check our :ref:`page on that<tests>`.
+
+Run kw without installing it
+----------------------------
+Apart from using the installed version of kw, you can also use your local
+version by going into it and running `./kw` with any other option you want,
+just like you would do when you run kw that is installed in your machine.
+
+For instance, suppose you changed the `help.sh` file and want to see if
+everything's working as expected. Inside the kw directory you're working on,
+you can just invoke::
+
+  ./kw help
+
+This a convenient option when you're developing for kw and don't want to install
+it every time you make a new change to see the result.
+
+Documentation
+-------------
+As important as writing tests is documenting things. Our documentation is
+located in the `documentation` folder.
+
+Whether you changed the way a command behaves or added a new feature, modify
+the documentation accordingly and preferably with examples. Also, in the source
+code, we tend to comment on top of each function, explaining its usage.
+
+ShellCheck and shfmt
+--------------------
+kw follows its own :ref:`code style guide<coding-style>`, and makes use of
+tools, such as ShellCheck and :ref:`shfmt<shfmt-label>`, to make sure this code
+style is being followed.
+
+Git pre-commit hook
+===================
+A very handy way to ensure that our code is well-formatted and follows our
```

Comment:
s/ensure that our code is/ensure that your code is/

hooks are something on the developer side, replace 'our' by 'your'.